lib/order.rb

```diff
@@ -0,0 +1,81 @@
+require 'csv' #ruby reads the csv file
+require_relative 'customer' #knows about customer class
+
+#assume there is only one of each product
+class Order
+
+  attr_reader :id
+  attr_accessor :products,  :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id #number
+    @products = products #{"melon" => 9.99, "chips" => 3.00} ] cost from cvs?
+    @customer = customer #from customer class
+    @fulfillment_status = fulfillment_status
+
+    raise ArgumentError.new("Invalid fulfillment status.") if [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status) != true
+  end
+
+  def total
+    if @products.empty?
+      return 0
+    else
+      total = @products.values.sum
+      total = (total*1.075).round(2)
+    end
+    return total
+  end
+
+  def add_product(product_name, product_price)
+    raise ArgumentError.new("This product already exists.") if  @products.include?(product_name)
+
+    @products[product_name] = product_price
+    # data = {product_name => product_price}  foam[:500]
+    # @products << data
+
+  end
+
+  # def remove_product(product_name)
+  # raise ArgumentError.new("Product not found.") if product_name == nil
+  #if @products.key?(name)
+  # # return @products.delete(name)
+  # end
+  # write a test for it it is similar to the add_product test
+  # "mellow yellow".split("ello")   #=> ["m", "w y", "w"]
+  def self.all
+    #returns a collection of Order instances representing all of the Orders
+    # described in the CSV file
+    #all = []
+    # id, products, customer, fulfillment_status
+    orders = CSV.read('data/orders.csv').map do |order_data| #{|x| x + "!"} array.map { |string| string.upcase }
+
+    id = order_data[0].to_i
+    products = product_hash[order_data[1]] #invoke helper method to parse mirror what csv shows contains product and price
```

Comment:
Code-style wise..... Ruby doesn't force us to use parentheses, but whenever I invoke a method, it helps me to add parentheses.

I might suggest refactoring my code to look more like

```suggestion
    products = product_hash(order_data[1])
```